run/O2HitMerger.h

```diff
@@ -191,7 +192,7 @@ class O2HitMerger : public FairMQDevice
   void fillBranch(int eventID, std::string const& name, T* ptr)
   {
     // fetch tree into which to fill
-
+    const std::lock_guard<std::mutex> lock(mMapsMtx);
```

Comment:
@shahor02 : Do you know at what line there is a race condition?

Reply:
I've seen the mEventToTTreeMap changing its state between the ``mEventToTTreeMap.find`` and ``mEventToTMemFileMap[eventID] = new ``, I assume another thread was modifying it at the same time (which may happen on any call mEventToTMemFileMap[eventID]  with new eventID). The whole construct seems to be somewhat shaky: this map is accessed in multiple places. I am wondering if it is not safer to pass to the merger the number of requested events to simulate so that it allocates from the beginning a container of final size.

I also left tree filling under the mutex as suspected that they are clashing and that this might be related to problems in the destructor. 

But my changes did not fully eliminate the crashes, so there should be something more.

container is not guaranteed at